lib/ASTGen/Sources/ASTGen/CompilerBuildConfiguration.swift

```diff
@@ -290,3 +309,53 @@ public func freeConfiguredRegions(
 ) {
   UnsafeMutableBufferPointer(start: regions, count: numRegions).deallocate()
 }
+
+/// Evaluate the #if condition at ifClauseLocationPtr.
+@_cdecl("swift_ASTGen_evaluatePoundIfCondition")
+public func evaluatePoundIfCondition(
+  astContext: BridgedASTContext,
+  diagEnginePtr: UnsafeMutableRawPointer,
+  sourceFileBuffer: BridgedStringRef,
+  ifConditionText: BridgedStringRef,
+  shouldEvaluate: Bool
+) -> Int {
+  // Retrieve the #if condition that we're evaluating here.
+  // FIXME: Use 'ExportedSourceFile' when C++ parser is replaced.
+  let textBuffer = UnsafeBufferPointer<UInt8>(start: ifConditionText.data, count: ifConditionText.count)
```

Comment:
Don’t we have an `ExportedSourceFile` already (the one that we also use for macro evaluation)? Or is this happening earlier where we haven’t created the `ExportedSourceFile` yet?

Reply:
I'm following the path @rintaro used for checking the definitions of macros, which avoids running the swift-syntax parser on the whole file (for performance reasons). This also avoids the somewhat brittle "go find this exact swift-syntax node at this source location" lookup we need to do with that approach, which can fail when the C++ parser and SwiftParser don't recover in exactly the same way.

Reply:
I see. I was still thinking we had the syntax tree for the entire file anyway and forgot that we no longer parse it completely.